server/src/main/java/org/elasticsearch/action/admin/indices/get/GetIndexRequest.java

```diff
@@ -149,6 +150,24 @@ public boolean humanReadable() {
         return humanReadable;
     }
 
+    public GetIndexRequest flatSettings(boolean flatSettings) {
+        this.flatSettings = flatSettings;
+        return this;
+    }
+
+    public boolean flatSettings() {
+        return flatSettings;
+    }
+
+    public GetIndexRequest includeDefaults(boolean includeDefaults) {
+        this.includeDefaults = includeDefaults;
+        return this;
+    }
+
+    public boolean includeDefaults() {
```

Comment:
nit: could you add some javadocs to these new methods, also explaining that this are only used by the high-level REST client?

Reply:
Sure! I have used wording from [https://github.com/elastic/elasticsearch/blob/master/rest-api-spec/src/main/resources/rest-api-spec/api/indices.exists.json](here), since my (lack of) knowledge of Elasticsearch would most probably result in something which is not so precise.